sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeletedKey.cs

```diff
@@ -3,27 +3,66 @@
 // license information.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Text.Json;
 
 namespace Azure.Security.KeyVault.Keys
 {
     public class DeletedKey : Key
     {
         public string RecoveryId { get; private set; }
 
-        public DateTime? DeletedDate { get; private set; }
+        public DateTimeOffset? DeletedDate { get; private set; }
 
-        public DateTime? ScheduledPurgeDate { get; private set; }
+        public DateTimeOffset? ScheduledPurgeDate { get; private set; }
 
-        public DeletedKey(string name, string recoveryId, DateTime? deletedDate, DateTime? scheduledPurge)
+        public DeletedKey(string name) : base(name) { }
+
+        public DeletedKey (string name, string recoveryId, DateTimeOffset? deletedDate, DateTimeOffset? scheduledPurge)
             : base(name)
         {
             RecoveryId = recoveryId;
             DeletedDate = deletedDate;
             ScheduledPurgeDate = scheduledPurge;
         }
+
+        internal override void WriteProperties(ref Utf8JsonWriter json)
```

Comment:
The latest preview changed `Utf8JsonWriter` to a class so passing it by ref shouldn't be necessary anymore. Just an fyi.